dlib/serialize.h

```diff
@@ -983,6 +1224,144 @@ namespace dlib
 
 // ----------------------------------------------------------------------------------------
 
+    template <typename domain, typename range, typename hash, typename keyEqual, typename alloc>
+    void serialize (
+        const std::unordered_map<domain, range, hash, keyEqual, alloc>& item,
+        std::ostream& out
+    )
+    {
+        try
+        { 
+            const unsigned long size = static_cast<unsigned long>(item.size());
+            serialize(size,out); 
+            for (const auto& x : item)
+            {
+                serialize(x.first,out);
+                serialize(x.second,out);
+            }
+        }
+        catch (serialization_error& e)
+        { throw serialization_error(e.info + "\n   while serializing object of type std::unordered_map"); }
+    }
+
+    template <typename domain, typename range, typename hash, typename keyEqual, typename alloc>
+    void deserialize (
+        std::unordered_map<domain, range, hash, keyEqual, alloc>& item,
+        std::istream& in
+    )
+    {
+        try 
+        { 
+            item.clear();
+            unsigned long size;
```

Comment:
And then here `size_t size;`

Reply:
Okidok. Again, I was following what you'd done in other areas like ```std::map```

Reply:
Long long ago I used to run into a compiler that would fail to compile when it saw size_t. That’s why it’s not as used as it should be.

Reply:
Let me guess, windows